src/core/solver.py

```
@@ -272,69 +285,210 @@ def add_scoring_objectives(self, strategy: ScoringStrategy) -> None:
+    def add_solution_hints(self) -> None:
+        """Applies previous assignments as solver hints for warm starts."""
+        if not self.cfg.hints_by_fingerprint and not self.cfg.hints_by_index:
+            return
+
+        hinted_groups: dict[int, int] = {}
+        for p_idx, p in enumerate(self.participants):
+            group_id = None
+            if self.cfg.hints_by_fingerprint:
+                group_id = self.cfg.hints_by_fingerprint.get(p.fingerprint)
+            if group_id is None and self.cfg.hints_by_index:
+                if p.original_index is not None:
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)